provider/ejp.py

```
@@ -76,22 +72,20 @@ def get_s3key(self, s3_key_name, bucket=None):
 
         return s3key
 
+    def write_content_to_file(self, filename, content, mode="wb"):
+        "write the content to a file in the tmp_dir"
+        document = os.path.join(self.get_tmp_dir(), filename)
+        with open(document, 'wb') as fp:
+            fp.write(content)
+            return document
+        # default if it does not work
+        return None
```

silent failure? Would an exception be more explicit if a file cannot be written?

---

although I think the `return document` won't be reached only in case an exception is raised before it, so this `return None` line may be unreachable code

---

I did create that function quickly to get tests to pass and to make writing content to disk more general purpose after removing the filesystem provider. I added some more exceptions and some specific tests for it. I hope it is an improvement!